hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java

```diff
@@ -255,17 +276,49 @@ final var record =
         }
     }
 
+    /**
+     * When we do an airdrop we need to check if there are custom fees that needs to be paid by the receiver. 
+     * If there are, an error is returned.
+     * However, there is an exception to this rule - if the receiver is the fee collector or the treasury account 
+     * they are exempt from paying the custom fees thus we don't need to check if there are custom fees.
+     * This method returns if the receiver is the fee collector or the treasury account.
+     */
+    private static boolean skipCustomFeeValidation(
+        List<CustomFee> customFees,
+        AccountID receiverId,
+        AccountID treasuryId) {
+        var theReceiverIsTheTreasury = receiverId.equals(treasuryId);
+        return thereIsASingleFeeCollectorAndItIsTheReceiver(customFees, receiverId) || theReceiverIsTheTreasury;
```

Comment:
as a micro-optimization consider reversing the two sides of the `||`

Reply:
It turns out that we have a method that already does the same thing. I should've known - it's not a new functionality...
Anyway, I've deleted my methods and I'm calling `CustomFeeExemptions.isPayerExempt` instead.